inc/cookie/namespace.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace HM\Delegated_Auth\Cookie;
+
+use function HM\Delegated_Auth\synchronize_user_for_token;
+use WP_Error;
+
+function is_enabled() : bool {
+	return defined( 'HM_DELEGATED_AUTH_CLIENT_ID' );
+}
+/**
+ * Login page footer for the login link.
+ */
+function on_login_form() : void {
+	?>
+	<p><a href="<?php echo esc_url( get_authorize_url() ); ?>">Login with Delegated Auth</a></p>
+	<?php
+}
+
+function on_load() : void {
+	if ( wp_parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH ) !== '/hm-delegated-auth-callback' ) {
+		return;
+	}
+
+	$code = sanitize_text_field( $_GET['code'] );
+	$auth = on_auth_callback( $code );
+	if ( is_wp_error( $auth ) ) {
+		wp_die( $auth->get_error_message() );
```

Comment:
You can pass the error instance instead (it will also use a title from the error's data if it exists)